client/src/core/ID.res

```diff
@@ -18,6 +18,9 @@ module T = {
   let generate = (): t => fromInt(Js_math.random_int(0, 2147483647))
 }
 
+// WHAT why do we need these custom Set and Map modules here?
+// Some odd ReScript limitation thing?
```

Comment:
OCaml/rescript does not have comparison traits like F#, so Sets and Tables need comparators added to them (There is a polymorphic comparison, used for `>`, etc, but it's not always safe to use.

This is frustrating for basic types, so Belt, Tablecloth, Jane st, etc, all add default Sets and Maps for int and string keys. If you want to have an easy to use Map or Set for other keys though, you need to define one.

However you only need a subset of functions, specifically the set where the type is not implicit from the arguments.